### PR TITLE
Feat/rk4

### PR DESCRIPTION
**Pull request by staslist**

### CODE REVIEW

**Comment by hkashyap** [comment]
Hi Stas, the CARLsim3 "firingRateCPUandGPU_EULERvsRUNGE_KUTTA" test in coba.cpp is missing. That or an equivalent test should be included. Is there any more RK4 test that you wrote for CARLsim3?

**Comment by staslist** [comment]
I've added the CUBA tests. I do not have data necessary to write TEST(COBA, firingRateVsData). If nobody minds I'd like to merge and move onto compartment model.

File: carlsim/test/cuba.cpp [hunk 1]
```diff
@@ -152,66 +153,208 @@ TEST(CUBA, firingRateCPUvsGPU) {
...
+		sim->runNetwork(0, 100);
+		sim->setExternalCurrent(g1, 70);
+		sim->runNetwork(0, 900);
+		spkMonG1->stopRecording();
+
+		//Data is taken from page 275 of Dynamical Systems in Neuroscience The Geometry of Excitability and Bursting
+		//by Eugene M. Izhikevich
+		EXPECT_FLOAT_EQ(spkMonG1->getPopMeanFiringRate(), 6.0f);
+		std::vector<std::vector<int> > spikeTimes = spkMonG1->getSpikeVector2D();
+		EXPECT_GT(spikeTimes[0][0], 195);
```
Comment (tingshuc) [inline carlsim/test/cuba.cpp]: Spike times should be fixed given no randomness. Use EXPECT_EQ() instead. The timing maybe a little bit off on different platforms.

File: carlsim/test/cuba.cpp [hunk 2]
```diff
@@ -152,66 +153,208 @@ TEST(CUBA, firingRateCPUvsGPU) {
...
+		sim->runNetwork(0, 500);
+		sim->setExternalCurrent(g1, 0);
+		sim->runNetwork(0, 400);
+		spkMonG1->stopRecording();
+
+		//Data is taken from page 275 of Dynamical Systems in Neuroscience The Geometry of Excitability and Bursting
+		//by Eugene M. Izhikevich
+		EXPECT_FLOAT_EQ(spkMonG1->getPopMeanFiringRate(), 5.0f);
+		std::vector<std::vector<int> > spikeTimes = spkMonG1->getSpikeVector2D();
+		EXPECT_GT(spikeTimes[0][0], 100);
```
Comment (tingshuc) [inline carlsim/test/cuba.cpp]: Spike times should be fixed given no randomness. Use EXPECT_EQ() instead. The timing maybe a little bit off on different platforms.

**Pull request merged**